modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/service/test/AccountEntryLocalServiceTest.java

```diff
@@ -199,13 +200,25 @@ public void testAccountEntryObjectValidationRule() throws Exception {
 			Assert.fail();
 		}
 		catch (ModelListenerException modelListenerException) {
-			String message = modelListenerException.getMessage();
+			ObjectValidationRuleEngineException
+				objectValidationRuleEngineException =
+					(ObjectValidationRuleEngineException)
+						modelListenerException.getCause();
 
-			Assert.assertTrue(message.contains("This name is invalid."));
+			ObjectValidationRuleResult objectValidationRuleResult =
+				objectValidationRuleEngineException.
+					getObjectValidationRuleResults(
+					).get(
+						0
+					);
+
+			String errorMessage = objectValidationRuleResult.getErrorMessage();
+
+			Assert.assertTrue(errorMessage.contains("This name is invalid."));
 
 			Assert.assertTrue(
 				modelListenerException.getCause() instanceof
-					ObjectValidationRuleEngineException.InvalidFields);
+					ObjectValidationRuleEngineException);
```

Comment:
Can we move this before the variable creation `objectValidationRuleEngineException` ?